compiler/ballerina-lang/src/main/java/io/ballerina/projects/Package.java

```diff
@@ -256,6 +256,48 @@ public CodeGeneratorResult runCodeGeneratorPlugins() {
         return codeGeneratorManager.runCodeGenerators(this);
     }
 
+    /**
+     * Run {@code CodeModifier} tasks in engaged {@code CompilerPlugin}s.
+     * <p>
+     * Returns a new package instances with modified files and a collected diagnostics
+     * reported by the code modifier tasks in form of a {@code CodeModifierResult} instance.
+     * <p>
+     * Here is a sample usage of this API: <pre>
+     *   Project project = BuildProject.load(Paths.get(...));
+     *   Package currentPackage = project.currentPackage();
+     *   Package packageWithGenFiles = currentPackage.runCodeModifierPlugins();
+     *
+     *   // Compile the package with generated files.
+     *   PackageCompilation compilation = packageWithGenFiles.getCompilation();
+     *   </pre>
+     * <p>
+     * This method does not run other tasks such as {@code CodeAnalyzer}s in engaged compiler plugins.
+     *
+     * @return a {@code CodeModifierResult} instance
+     */
+    public CodeModifierResult runCodeModifierPlugins() {
```

Comment:
We should try to run code modifiers and code generators in one go. Running these separately will compile the files multiple times which will increase the build time.